opm/parser/eclipse/EclipseState/Schedule/Action/ActionResult.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright 2019 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ACTION_RESULT_HPP
+#define ACTION_RESULT_HPP
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace Opm {
+namespace Action {
+
+/*
+  The Action::Result class is used to hold the boolean result of a ACTIONX
+  condition like:
+
+        WWCT > 0.75
+
+   and also the final evaluation of an ACTIONX condition comes as a
+   Action::Result instance. In addition to a boolean member the Action::Result
+   class keeps track of the wells matching the conditions in the ACTIONX keyword.
+
+   The set of matching wells will be passed to the Schedule::applyAction() if
+   the condition evaluates to true.
+*/
+
+
+class WellSet {
+public:
+    WellSet();
+    WellSet(const std::vector<std::string>& wells);
+    void add(const std::string& well);
+
+    std::size_t size() const;
+    bool scalar() const;
+    std::vector<std::string> wells() const;
+    bool contains(const std::string& well) const;
+
+    WellSet& intersect(const WellSet& other);
+    WellSet& add(const WellSet& other);
+private:
+    bool m_scalar;
+    std::unordered_set<std::string> well_set;
+};
+
+
+
+class Result {
+public:
+    explicit Result(bool result_arg);
+    Result(bool result_arg, const std::vector<std::string>& wells);
+    Result(bool result_arg, const WellSet& wells);
+
+    bool result() const;
```

Comment:
I think the result() and scalar() members are quite confusing.

As I understand the PR description, the (or one) problem is dealing with expressions like `FWPR > 1000` which generate a result with an empty set of wells currently. The simple fix would then be to have such expressions generate a result object containing all wells. That way, a Result is "the set of wells for which the conditions hold" always.

A possible optimization would be to have a special flag for "all wells". I think that is what you have done here, but I do not understand why this state seems to be called "scalar". Furthermore I think calling wells() will not give the correct set of wells (all of them...) for such a Result.

The result() member is also confusing. I'll assume the member is supposed to be understood like this: If result() is `true`, then I contain all wells for which the conditions hold, if result() is false, then I contain all wells for which the conditions do not hold. Then again the wells() member is suspect, or at least hard to use: its meaning depends on the value of result(). I think a Result should always behave as if result() is true, if you need to negate it I would rather have a `not` operator doing that.